### ISSUE TITLE
acceptance: TestGossipRestart is flaky

### ISSUE THREAD

**Issue by tschottdorf** [cockroachdb/cockroach#3003]
The following test appears to have failed:

[#8911](https://circleci.com/gh/cockroachdb/cockroach/8911):

```
E1104 18:13:59.302041 15896 acceptance/localcluster/localcluster.go:398  node=1 status=restart
E1104 18:13:59.741464 15896 acceptance/localcluster/localcluster.go:398  node=2 status=start
I1104 18:14:01.462938 15896 acceptance/gossip_peerings_test.go:153  waiting for gossip to be connected
E1104 18:14:01.465957 15896 acceptance/localcluster/localcluster.go:398  node=2 status=restart
I1104 18:14:05.684653 15896 acceptance/localcluster/localcluster.go:525  stopping
--- FAIL: TestGossipRestart (34.53s)
	testing.go:157: acceptance/gossip_peerings_test.go:62: condition failed to evaluate within 1s: acceptance/gossip_peerings_test.go:57: node 0: acceptance/gossip_peerings_test.go:85: sentinel not found
=== RUN   TestPut
I1104 18:14:10.805691 15896 acceptance/localcluster/localcluster.go:211  started docker-spy: 172.17.0.24
I1104 18:14:10.805719 15896 acceptance/localcluster/localcluster.go:218  initializing cluster
I1104 18:14:13.497202 15896 acceptance/localcluster/localcluster.go:328  creating ca (1024bit) in: /home/ubuntu/cockroach/.localcluster.certs.516647497
I1104 18:14:13.528040 15896 acceptance/localcluster/localcluster.go:333  creating node (1024bit) certs in: /home/ubuntu/cockroach/.localcluster.certs.516647497
I1104 18:14:15.910391 15896 acceptance/localcluster/localcluster.go:381  *** started roach0.local ***
  ui:    https://0.0.0.0:32784
  trace: https://0.0.0.0:32784/debug/requests
  logs:  /tmp/circle-artifacts.WdIem3U/acceptance/roach0.local/cockroach.INFO
--
I1104 18:15:29.985383 15896 acceptance/status_server_test.go:67  OK response from https://0.0.0.0:32798/_status/logs/3
I1104 18:15:29.989537 15896 acceptance/status_server_test.go:67  OK response from https://0.0.0.0:32798/_status/stacks/3
I1104 18:15:29.992682 15896 acceptance/status_server_test.go:67  OK response from https://0.0.0.0:32798/_status/nodes/3
I1104 18:15:30.093066 15896 acceptance/localcluster/localcluster.go:525  stopping
--- PASS: TestStatusServer (16.64s)
FAIL
ok github.com/cockroachdb/cockroach/acceptance 1337s

```
Please assign, take a look and update the issue accordingly.

**Comment by tschottdorf** [comment]
log from node 1 (after it's restarted):
```
I1104 18:13:21.191920 1 storage/engine/rocksdb.go:103  opening rocksdb instance at "/data0"
I1104 18:13:21.195145 1 gossip/gossip.go:666  starting client to roach1.local:26257
I1104 18:13:21.205681 1 server/node.go:286  initialized store store=1:1 ([ssd]=/data0): {Capacity:257668677632 Available:180744536064 RangeCount:0}
I1104 18:13:21.205790 1 server/node.go:213  node ID 1 initialized
I1104 18:13:21.206053 1 gossip/gossip.go:175  gossiping node descriptor node_id:1 address:<network_field:"tcp" address_field:"roach0.local:26257" > attrs:<> 
I1104 18:13:21.206278 1 server/node.go:376  connecting to gossip network to verify cluster ID...
I1104 18:13:21.207183 1 server/node.go:393  node connected via gossip and verified as part of cluster "9df9c1d7-afbc-4f91-ad89-a7b24ce4c503"
I1104 18:13:21.207335 1 server/node.go:251  Started node with [[ssd]=/data0] engine(s) and attributes []
I1104 18:13:21.207473 1 server/server.go:202  starting https server at roach0.local:26257
W1104 18:13:22.696480 1 gossip/gossip.go:594  missing sentinel gossip 1; assuming partition and reconnecting
I1104 18:13:22.698088 1 gossip/gossip.go:666  starting client to roach2.local:26257
W1104 18:13:24.472348 1 gossip/gossip.go:594  missing sentinel gossip 1; assuming partition and reconnecting
W1104 18:13:26.195177 1 gossip/gossip.go:643  connected to gossip but missing sentinel. Has the cluster been initialized? Use "cockroach init" to initialize.
W1104 18:13:26.832465 1 gossip/gossip.go:594  missing sentinel gossip 1; assuming partition and reconnecting
I1104 18:13:27.364170 1 gossip/client.go:80  gossip client to roach1.local:26257: unexpected EOF
W1104 18:13:27.364582 1 gossip/gossip.go:594  missing sentinel gossip 1; assuming partition and reconnecting
I1104 18:13:27.367423 1 gossip/gossip.go:666  starting client to roach1.local:26257
W1104 18:13:27.368801 1 rpc/client.go:226  dial tcp 172.17.0.12:26257: getsockopt: connection refused
E1104 18:13:27.505270 1 server/raft_transport.go:175  raft message to node 2 failed: connection is shut down
E1104 18:13:27.506280 1 server/raft_transport.go:175  raft message to node 2 failed: connection is shut down
W1104 18:13:27.517282 1 rpc/client.go:234  connection is shut down
E1104 18:13:27.805187 1 server/raft_transport.go:175  raft message to node 2 failed: connection is shut down
E1104 18:13:28.105185 1 server/raft_transport.go:175  raft message to node 2 failed: connection is shut down
E1104 18:13:28.405230 1 server/raft_transport.go:175  raft message to node 2 failed: connection is shut down
E1104 18:13:28.705238 1 server/raft_transport.go:175  raft message to node 2 failed: connection is shut down
W1104 18:13:28.997255 1 gossip/gossip.go:594  missing sentinel gossip 1; assuming partition and reconnecting
E1104 18:13:29.005679 1 server/raft_transport.go:175  raft message to node 2 failed: connection is shut down
E1104 18:13:29.305187 1 server/raft_transport.go:175  raft message to node 2 failed: connection is shut down
E1104 18:13:29.605206 1 server/raft_transport.go:175  raft message to node 2 failed: connection is shut down
E1104 18:13:29.905244 1 server/raft_transport.go:175  raft message to node 2 failed: connection is shut down
E1104 18:13:30.205270 1 server/raft_transport.go:175  raft message to node 2 failed: connection is shut down
E1104 18:13:30.205844 1 server/raft_transport.go:175  raft message to node 2 failed: connection is shut down
E1104 18:13:30.489971 1 server/raft_transport.go:175  raft message to node 2 failed: connection is shut down
E1104 18:13:30.505142 1 server/raft_transport.go:175  raft message to node 2 failed: connection is shut down
E1104 18:13:30.779461 1 server/raft_transport.go:175  raft message to node 2 failed: connection is shut down
E1104 18:13:30.805164 1 server/raft_transport.go:175  raft message to node 2 failed: connection is shut down
E1104 18:13:31.079426 1 server/raft_transport.go:175  raft message to node 2 failed: connection is shut down
E1104 18:13:31.105193 1 server/raft_transport.go:175  raft message to node 2 failed: connection is shut down
I1104 18:13:31.208481 1 server/status/runtime.go:122  runtime stats: 58 goroutines, 3.24MiB active, 0.00cgo/sec, 0.00/0.00 %(u/s)time, 0.00 %gc (2x)
I1104 18:13:31.208752 1 storage/store.go:1692  system config not yet available
W1104 18:13:31.319918 1 rpc/client.go:226  dial tcp 172.17.0.12:26257: i/o timeout
E1104 18:13:31.379423 1 server/raft_transport.go:175  raft message to node 2 failed: connection is shut down
E1104 18:13:31.405327 1 server/raft_transport.go:175  raft message to node 2 failed: connection is shut down
W1104 18:13:31.425522 1 gossip/gossip.go:594  missing sentinel gossip 1; assuming partition and reconnecting
W1104 18:13:31.504511 1 rpc/client.go:226  dial tcp 172.17.0.12:26257: i/o timeout
```

**Comment by tschottdorf** [comment]
closing for #9372
